### DISCUSSION TITLE
Different Colored Text Appearing in the Same Text Box

### DISCUSSION THREAD

**Discussion by r-shand** [TomSchimansky/CustomTkinter#1369] [Q&A]
Hey, I was wondering if it was possible to have different colored text in ctk text boxes like it is with normal tk textboxes.

I have this method in tkinter that adds text in the textbox with the specified color. 

def display_text(self, text, color="#FFFFFF"):
        start_index = self.output_widget.index(tk.END + "-1l")
        # Insert the text
        self.output_widget.insert(tk.END, text + '\n')
        # Create a tag for the specified color, if not already created
        self.output_widget.tag_configure(color, foreground=color)
        # Apply the color tag to the inserted text
        end_index = self.output_widget.index(tk.END + "-1c")
        self.output_widget.tag_add(color, start_index, end_index)
        # Scroll to the end
        self.output_widget.see(tk.END)

I am trying to adapt this for customtkinter but there is no tag_configure and the attribute I need to change is now "text_color" but this will change all of the text in the text box to that color even though I just wanted it to be the one new message printing in the custom color. (below is my code for custom tkinter)

def display_text(self, text, color="#FFFE00"):
        start_index = self.output_widget.index(tk.END + "-1l")
        # Insert the text
        self.output_widget.insert(tk.END, text + '\n')
        self.output_widget.configure(color, text_color=color)
        end_index = self.output_widget.index(tk.END + "-1c")
        self.output_widget.tag_add(color, start_index, end_index)
        # Scroll to the end
        self.output_widget.see(tk.END)

Any insight is appreciated! Thanks

**Comment by Akascape** [comment]
@r-shand Yes, you can use `tag_add` and `tag_config` to change text foreground colors of ctktextbox. Both ctktextbox and text widget of tkinter works the same.

**Reply by r-shand** [reply]
Thank you so much! I was trying to use tag_configure and not tag_config
